illumos-utils/src/zone.rs

```diff
@@ -833,15 +887,127 @@ impl Zones {
         };
 
         // Actually perform address allocation.
-        Self::create_address_internal(zone, addrobj, addrtype)?;
-
-        Self::get_address_impl(zone, addrobj)
+        Self::create_address_internal(executor, zone, addrobj, addrtype)?;
+        Self::get_address_impl(executor, zone, addrobj)
     }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
+    use crate::host::{FakeExecutor, Input, OutputExt, StaticHandler};
+    use omicron_test_utils::dev;
+    use std::process::Output;
+
+    #[tokio::test]
+    async fn install_new_zone_calls_config_then_install() {
```

Comment:
This test looks great, but gave me pause about how we might extend this to cover (e.g.) bigger sled-agent tasks. Naively, in order for crates that _use_ illumos-utils intermixed with their own `Command`s to create `StaticHandler`s, they will also need to inject the appropriate expectations for the calls illumos-utils makes on their behalf, which seems like it might be onerous.

I'm not sure this is an actual problem, or if it is that there isn't a way to make it easier. Just thinking out loud.